hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -123,6 +144,9 @@ def depth(self):
     def index(self):
         return len(self.buffer)
 
+    def each_block_bounds(self):
+        return (block.bounds for block in self.blocks)
```

Comment:
I'd be more comfortable returning a list than a generator here - slightly less efficient, but fewer ways to stuff up later.

Reply:
Originally the idea was to not need this method at all. It exists as a concession to the existing code sites that did `for u, v in data.blocks`, because I couldn't find a cleaner way to update them.

I have a slight preference for it being a generator, to encourage any new code to use `data.blocks` directly instead.

But I could go either way on it, and I can't deny that I *already* made one of the mistakes you hinted at, in a previous version of `update_shrink_target`.